src/Servers/Kestrel/Core/src/Internal/Infrastructure/StringUtilities.cs

```diff
@@ -115,6 +115,93 @@ public static unsafe bool TryGetAsciiString(byte* input, char* output, int count
             return isValid;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        public static unsafe void GetLatin1String(byte* input, char* output, int count)
```

Comment:
Honestly I'd rather this logic not be in aspnetcore. If this is pri-0 for you, let the API review team know that having a `Encoding.Latin1` static property (see proposal at https://github.com/dotnet/corefx/issues/42686) is a necessity for you, and keep the optimized logic wholly within the runtime itself.